src/Tests/Microsoft.NET.Sdk.Razor.Tests/StaticWebAssets/ReadStaticWebAssetsManifestFileTest.cs

```diff
@@ -0,0 +1,346 @@
+            
+            discoveryPattern.ItemSpec.ShouldBeEquivalentTo(Path.Combine("AnotherClassLib", "wwwroot"));
+            discoveryPattern.GetMetadata(nameof(StaticWebAssetsManifest.DiscoveryPattern.Source)).ShouldBeEquivalentTo("AnotherClassLib");
+            discoveryPattern.GetMetadata(nameof(StaticWebAssetsManifest.DiscoveryPattern.ContentRoot)).ShouldBeEquivalentTo($"{contentRoot}");
+            discoveryPattern.GetMetadata(nameof(StaticWebAssetsManifest.DiscoveryPattern.BasePath)).ShouldBeEquivalentTo("_content/AnotherClassLib");
+            discoveryPattern.GetMetadata(nameof(StaticWebAssetsManifest.DiscoveryPattern.Pattern)).ShouldBeEquivalentTo("**");
+        }
+
+        [Fact]
+        public void ReturnsErrorwhenManifestDoesNotExist()
+        {
+            var errorMessages = new List<string>();
+            var buildEngine = new Mock<IBuildEngine>();
+            buildEngine.Setup(e => e.LogErrorEvent(It.IsAny<BuildErrorEventArgs>()))
+                .Callback<BuildErrorEventArgs>(args => errorMessages.Add(args.Message));
+
+            var task = new ReadStaticWebAssetsManifestFile
+            {
+                BuildEngine = buildEngine.Object,
+                ManifestPath = "nonexisting.staticwebassets.json"
+            };
+
+            // Act
+            var result = task.Execute();
+
+            // Assert
+            result.Should().Be(false);
+            errorMessages.Count.Should().Be(1);
+            errorMessages[0].Should().Be("Manifest file at 'nonexisting.staticwebassets.json' not found.");
+            task.Assets.Should().BeNull();
+            task.DiscoveryPatterns.Should().BeNull();
+            task.ReferencedProjectsConfiguration.Should().BeNull();
+        }
+
+        [Fact]
+        public void ReturnsErrorwhenManifestIsMalformed()
+        {
+            var errorMessages = new List<string>();
+            var buildEngine = new Mock<IBuildEngine>();
+            buildEngine.Setup(e => e.LogErrorEvent(It.IsAny<BuildErrorEventArgs>()))
+                .Callback<BuildErrorEventArgs>(args => errorMessages.Add(args.Message));
+
+            var manifest = "{";
+            File.WriteAllText(TempFilePath, manifest);
+            var task = new ReadStaticWebAssetsManifestFile
+            {
+                BuildEngine = buildEngine.Object,
+                ManifestPath = TempFilePath
+            };
+
+            // Act
+            var result = task.Execute();
+
+            // Assert
+            result.Should().Be(false);
+            errorMessages.Count.Should().Be(1);
+            task.Assets.Should().BeNull();
+            task.DiscoveryPatterns.Should().BeNull();
+            task.ReferencedProjectsConfiguration.Should().BeNull();
+        }
+
+        public void Dispose()
+        {
+            try
+            {
+                if (TempFilePath != null && File.Exists(TempFilePath))
+                {
+                    File.Delete(TempFilePath);
```

Comment:
The SDK tests leave these files around so you can see it. They write to the Intermediate dir, and use the same file name for each run, so the number of files doesn't grow. Perhaps we could consider doing the same thing here.

Reply:
I changed to use the TestContext and put the temp files in the temp path.